ast/misc.go

```diff
@@ -93,6 +93,18 @@ type AuthOption struct {
 	// TODO: support auth_plugin
 }
 
+// Restore implements Node interface.
+func (n *AuthOption) Restore(ctx *RestoreCtx) error {
```

Comment:
Restore "IDENTIFIED WITH" will get "IDENTIFIED BY" ?

Reply:
And there is no test to cover `AutoOption` ?

Reply:
@tiancaiamao 
1. According to `parser.y`, it looks like syntax `IDENTIFIED WITH string` has no effect:
https://github.com/pingcap/parser/blob/63e2db7dbaddf946dc28c17077251b217fd5da26/parser.y#L7302-L7319

2. To test `Restore` of `AuthOption` (and `UserSpec`, `UserIdentity`), and to follow the testing routine of other types, it should be promoted as a `Node` and add `Accept` for it. Is that suitable?